### DISCUSSION TITLE
actuatorSurface for modeling Nacelle and hub to  make the Thrust forces more realistic

### DISCUSSION THREAD

**Discussion by ssthakar** [turbinesFoam/turbinesFoam#365] [Ideas]
I was running some tests on the NTNU S826 test case and noticed that while I was able to match my Cp vs TSR curves pretty neatly with those of the experiment, the turbinesFOAM library was underpredicting the thrust loading. 
Delving deeper I came across a few papers that modeled the nacelle/nacelle+hub as an actuator surface.  I was wondering if something like this was possible with the current library. 
Here is the [paper](https://onlinelibrary.wiley.com/doi/abs/10.1002/we.2162) i mentioned

**Comment by petebachant** [comment]
It's not in the current implementation, but it might not be too much effort to achieve.

**Reply by ssthakar** [reply]
I was thinking of creating two classes, actuatorSurfaceElement(holds in geometric information like surface normal surface tangent and point location) and actuatorSurfaceSource (collection of elements), in a similar approach to the implementation of the actuatorLineElement and actuatorLineSource classes in your code. I have tinkered with openFOAM source code in the past, but not to this extent. Is this a feasible approach?  Thanks!

**Reply by petebachant** [reply]
Sounds like a reasonable place to start, though it may be simpler to start with `actuatorSurfaceSource` first, keeping the element data internal in vectors, then break the element into its own class only if it makes the code more understandable, since an element on its own probably won't be useful.
